objectron_dataset_detection.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+
+__all__ = ['Objectron']
+
+import os
+import sys
+import io
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Tuple, Dict
+from PIL import Image
+import numpy as np
+import pickle
+
+# torch+torchvision
+import torch as th
+import torch.nn as nn
+import torchvision.io as thio
+from torchvision import transforms
+
+import torch_xla.utils.tf_record_reader as tfrr
+from google.cloud import storage
+
+
+NUM_KEYPOINTS = 9
+
+
+def _glob_objectron(bucket_name: str, prefix: str):
+    client = storage.Client.create_anonymous_client()
+    blobs = client.list_blobs(bucket_name, prefix=prefix)
+    return [blob.name for blob in blobs]
+
+def decode(example, feature_names:List[str] = []):
+    w = example['image/width'].item()
+    h = example['image/height'].item()
+    points = example['point_2d'].numpy()
+    num_instances = example['instance_num'].item()
+    points = points.reshape(num_instances, NUM_KEYPOINTS, 3)
+    image_data = example['image/encoded'].numpy().tobytes()
+    image = Image.open(io.BytesIO(image_data))
+    npa = np.asarray(image)
+    out = {
```

Comment:
We might want to formalize the list of these keys into a constant so that we don't have hardcoded ames and constants everywhere ... also out of scope for this PR

Reply:
It’s a good idea because it’s the same key value everywhere.